homeassistant/components/tplink/switch.py

```diff
@@ -38,9 +39,48 @@ async def async_setup_entry(
     else:
         entities.append(SmartPlugSwitch(device, coordinator))
 
+    entities.append(SmartPlugLedSwitch(device, coordinator))
+
     async_add_entities(entities)
 
 
+class SmartPlugLedSwitch(CoordinatedTPLinkEntity, SwitchEntity):
+    """Representation of switch for the LED of a TPLink Smart Plug."""
+
+    coordinator: TPLinkDataUpdateCoordinator
+
+    def __init__(
+        self, device: SmartDevice, coordinator: TPLinkDataUpdateCoordinator
+    ) -> None:
+        """Initialize the LED switch."""
+        super().__init__(device, coordinator)
+
+        self._attr_entity_category = ENTITY_CATEGORY_CONFIG
+        self._attr_name = f"{device.alias} (LED)"
+        self._attr_unique_id = f"{legacy_device_id(device)}_led"
+        self._attr_icon = "mdi:led-on"
+
+    @property
+    def name(self) -> str:
+        """Return the name of the Smart Plug with appended LED mention."""
+        return cast(str, self._attr_name)
+
```

Comment:
```suggestion
```

Reply:
If not adding this, the name is skipped since this is overriden by one of the parent classes (`TPLinkDataUpdateCoordinator`). Alternative would be to avoid the coordinator from overriding that, and setting `_attr_name` in all the other subclasses. Which is preferable?